### PR TITLE
Sns

### PR DESCRIPTION
**Pull request by uk4z**
Created an SNS feature

### CODE REVIEW

File: src/routes/sns/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+						<button class="action-btn" on:click={notifyIcpDeposit}>Confirm SNS deposit</button>
+					{/if}
+				</div>
+				<div class="step-container" in:fade={{ duration: 500 }}>
+					<div class="instruction-container">
+						<div class="number-step-container">
+							<span class="round">3</span>
+						</div>
+						<span>Collect the minted nICP tokens to the SNS principal.</span>
+					</div>
```
Comment (leo-ship-it) [inline src/routes/sns/+page.svelte:204]:
```suggestion
						<span>Collect the minted nICP tokens to the SNS governance principal.</span>
					</div>
```

File: package.json [hunk 2]
```diff
@@ -40,11 +40,13 @@
 		"@dfinity/candid": "^1.3.0",
 		"@dfinity/identity": "^1.3.0",
 		"@dfinity/ledger-icp": "next",
+		"@dfinity/ledger-icrc": "^2.4.0",
```
Comment (leo-ship-it) [inline package.json:43]: unused deps

File: package.json [hunk 3]
```diff
@@ -40,11 +40,13 @@
...
 		"@dfinity/principal": "^1.3.0",
 		"@dfinity/utils": "next",
 		"@esbuild-plugins/node-globals-polyfill": "^0.2.3",
 		"@esbuild-plugins/node-modules-polyfill": "^0.2.2",
 		"@rollup/plugin-inject": "^5.0.5",
+		"crc": "^4.3.2",
```
Comment (leo-ship-it) [inline package.json:48]: Let's not use this package, use https://www.npmjs.com/package/@dfinity/ledger-icrc#gear-encodeicrcaccount instaed.

File: src/lib/stores.ts [hunk 4]
```diff
@@ -10,10 +10,14 @@ export const isConverting = writable<boolean>(false);
...
 export const inReceivingMenu = writable<boolean>(false);
 export const inMobileMenu = writable<boolean>(false);
+export const inSnsMenu = writable<boolean>(false);
 
 /* === Components === */
 export const language = writable<'en' | 'es' | 'ja' | 'ru'>('en');
 export const selectedAsset = writable<Asset>(new Asset(AssetType.ICP));
+export const selectedSns = writable<string>('BOOM DAO');
+export const snsPrincipal = writable<string>('xomae-vyaaa-aaaaq-aabhq-cai');
+export const snsHex = writable<string>('-/-');
```
Comment (leo-ship-it) [inline src/lib/stores.ts:20]:
```suggestion
export const encodedSnsIcrcAccount = writable<string>('-/-');
```

File: src/routes/sns/+page.svelte [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+					Uint8Array.from([...account.owner.toUint8Array(), ...account.subaccount[0]])
+				);
+				const crc32 = encodeBase32(crc);
+
+				const hex =
+					CANISTER_ID_BOOMERANG + '-' + crc32 + '.' + uint8ArrayToHexString(account.subaccount[0]);
+
+				if (hex !== previousHex) {
+					snsHex.set(hex);
+					previousHex = hex;
```
Comment (leo-ship-it) [inline src/routes/sns/+page.svelte:54]: All of that can be done with: https://www.npmjs.com/package/@dfinity/ledger-icrc#gear-encodeicrcaccount

File: package.json (hunk 3 above)
Comment (uk4z) [reply package.json]: Discussed irl

File: src/routes/sns/+page.svelte (hunk 5 above)
Comment (uk4z) [reply src/routes/sns/+page.svelte]: done.

File: src/lib/stores.ts (hunk 4 above)
Comment (uk4z) [reply src/lib/stores.ts]: done

**Pull request merged**